service/qos/service_level_controller.cc

```diff
@@ -88,15 +90,23 @@ future<> service_level_controller::drain() {
         _global_controller_db->dist_data_update_aborter.request_abort();
     }
     _global_controller_db->notifications_serializer.broken();
-    return std::exchange(_global_controller_db->distributed_data_update, make_ready_future<>()).then_wrapped([] (future<> f) {
-        try {
-            f.get();
-        } catch (const broken_semaphore& ignored) {
-        } catch (const sleep_aborted& ignored) {
-        } catch (const exceptions::unavailable_exception& ignored) {
-        } catch (const exceptions::read_timeout_exception& ignored) {
-        }
+
+    future<> data_accessor_stop_fut = make_ready_future<>();
+    if (_sl_data_accessor) {
+        data_accessor_stop_fut = _sl_data_accessor->stop();
+    }
+    return data_accessor_stop_fut.then([this] {
```

Comment:
I think that the order between stopping the dta_accessor and the distributed update should be switched. The reason is that once the distributed update loop is stopped, the are going to be very little (at current code state none) calls to the data_accessor as the distributed_update is one of it's only users.